src/app/shared/load-polygon-chart.directive.ts

```diff
@@ -91,5 +95,69 @@ export class LoadPolygonChartDirective implements OnChanges {
     g.append('path')
       .attr('d', areaBar(this.data))
       .style('fill', '#d9721f');
+
+    const focus = g.append('g').style('display', 'none');
+
+    focus.append('line')
+      .attr('id', 'focusLineX')
+      .attr('class', 'focusLine');
+
+    focus.append('circle')
+      .attr('id', 'focusCircle')
+      .attr('r', 1.5)
+      .attr('class', 'circle focusCircle');
+
+    focus.append('text')
+      .attr('id', 'focusText')
+      .attr('font-size', '10')
+      .attr('x', 1)
+      .attr('y', 8);
+
+    // @ts-ignore TS2345
+    const bisectDate = bisector((d, x) => +moment.utc(d[this.xkey]).isBefore(x)).right;
+
+    const rect = g.append('rect')
+      .attr('class', 'overlay')
+      .attr('width', this.width)
+      .attr('height', this.height)
+      .on('mouseover', () => focus.style('display', null))
+      .on('mouseout', () => focus.style('display', 'none'));
+
+    rect.on('mousemove', () => {
+      const coords = mouse(currentEvent.currentTarget);
+
+      const mouseDate: any = moment.utc(scaleX.invert(coords[0]));
+      // returns the index to the current data item
+      const i = Math.min(Math.max(bisectDate(this.appLoadPolygonChart, mouseDate), 0), this.appLoadPolygonChart.length - 1);
+      const d = this.appLoadPolygonChart[i];
+
+      const x = scaleX(isoParse(d[this.xkey]));
```

Comment:
Could you please double check if it works in such way? We have unix timestamp instead of ISO datetime now.